### PR TITLE
🚨 Fix remaning warnings for clang and gcc

### PR DESCRIPTION
**Pull request by AJenbo**

### CODE REVIEW

File: Source/codec.h [hunk 1]
```diff
@@ -10,7 +10,7 @@
 namespace devilution {
 
 int codec_decode(BYTE *pbSrcDst, DWORD size, const char *pszPassword);
-DWORD codec_get_encoded_len(DWORD dwSrcBytes);
-void codec_encode(BYTE *pbSrcDst, DWORD size, int size_64, const char *pszPassword);
+size_t codec_get_encoded_len(size_t dwSrcBytes);
```
Comment (glebm) [inline Source/codec.h:13]: This requires a `<cstddef>` include and should be `std::size_t`

File: Source/items.cpp [hunk 2]
```diff
@@ -1773,7 +1777,7 @@ void SaveItemPower(int i, item_effect_type power, int param1, int param2, int mi
 		break;
 	case IPL_DOPPELGANGER:
 		items[i]._iDamAcFlags |= 16;
-		// no break
+		// fall through
```
Comment (glebm) [inline Source/items.cpp:1780]: Use the C++17 attribute here and below:

```suggestion
		[[fallthrough]];
```

File: Source/mpqapi.cpp [hunk 3]
```diff
@@ -458,8 +458,6 @@ static int mpqapi_get_hash_index(int index, int hash_a, int hash_b, int locale)
 			continue;
 		if (cur_archive.sgpHashTbl[idx].hashcheck[1] != hash_b)
 			continue;
-		if (cur_archive.sgpHashTbl[idx].lcid != locale)
```
Comment (glebm) [inline Source/mpqapi.cpp:461]: Is this purely a warning fix?

File: Source/mpqapi.h [hunk 4]
```diff
@@ -28,7 +28,7 @@ struct _FILEHEADER {
 struct _HASHENTRY {
 	Uint32 hashcheck[2];
 	Uint32 lcid;
-	Uint32 block;
+	int32_t block;
```
Comment (glebm) [inline Source/mpqapi.h:31]: This changed signedness, was this intended?

File: Source/mpqapi.cpp (hunk 3 above)
Comment (AJenbo) [reply Source/mpqapi.cpp:461]: Unused, the easiest way to fix the related warning was to remove it.

File: Source/mpqapi.h (hunk 4 above)
Comment (AJenbo) [reply Source/mpqapi.h:31]: Yes, it is checked for -1 and -2

**Review by glebm — APPROVED** [review]

**Comment by AJenbo** [comment]
Now we just need a brave sole willing to deal with MSVC

**Pull request merged**